### PR TITLE
Allow empty media query list

### PR DESCRIPTION
**Pull request by upsuper**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14159)

### CODE REVIEW

File: components/style/media_queries.rs [hunk 1]
```diff
@@ -269,17 +275,16 @@ pub fn parse_media_query_list(input: &mut Parser) -> MediaList {
...
+    }
 }
 
 impl MediaList {
     pub fn evaluate(&self, device: &Device) -> bool {
         let viewport_size = device.au_viewport_size();
 
-        // Check if any queries match (OR condition)
-        self.media_queries.iter().any(|mq| {
+        // Check if it is an empty media query list or any queries match (OR condition)
```
Comment (upsuper) [inline components/style/media_queries.rs]: I don't think Servo currently exposes MediaList in DOM... CSSOM isn't the right spec to link here.

If we really want a link to spec, it should probably be [Media Queries 4](https://drafts.csswg.org/mediaqueries-4/#mq-list).

Comment (Wafflespeanut) [inline components/style/media_queries.rs]: Whoops.

**Pull request merged**